x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_form/relevance_tuning_item_content/boosts/boosts.tsx

```diff
@@ -0,0 +1,131 @@
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageContentBody,
+  EuiTitle,
+  EuiSuperSelect,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { TEXT } from '../../../../../../shared/constants/field_types';
+import { SchemaTypes } from '../../../../../../shared/types';
+
+import { BoostIcon } from '../../../boost_icon';
+import { FUNCTIONAL_DISPLAY, PROXIMITY_DISPLAY, VALUE_DISPLAY } from '../../../constants';
+import { RelevanceTuningLogic } from '../../../relevance_tuning_logic';
+import { Boost, BoostType } from '../../../types';
+
+import { BoostItem } from './boost_item';
+
+import './boosts.scss';
+
+const BASE_OPTIONS = [
+  {
+    value: 'add-boost',
+    inputDisplay: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.boosts.addBoostDropDownOptionLabel',
+      {
+        defaultMessage: 'Add boost',
+      }
+    ),
+    disabled: true,
+  },
+  {
+    value: BoostType.Functional,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Functional} />
+        {FUNCTIONAL_DISPLAY}
+      </>
+    ),
+  },
+  {
+    value: BoostType.Proximity,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Proximity} />
+        {PROXIMITY_DISPLAY}
+      </>
+    ),
+  },
+  {
+    value: BoostType.Value,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Value} />
+        {VALUE_DISPLAY}
+      </>
+    ),
+  },
+];
+
+const filterInvalidOptions = (value: BoostType, type: SchemaTypes) => {
+  // Proximity and Functional boost types are not valid for text fields
+  if (type === TEXT && [BoostType.Proximity, BoostType.Functional].includes(value)) return false;
+  return true;
+};
+
+interface Props {
+  name: string;
+  type: SchemaTypes;
+  boosts?: Boost[];
+}
+
+export const Boosts: React.FC<Props> = ({ name, type, boosts = [] }) => {
+  const { addBoost } = useActions(RelevanceTuningLogic);
+
+  const addBoostClick = (value: BoostType) => {
+    addBoost(name, value);
+  };
+
+  const selectOptions = useMemo(
+    () => BASE_OPTIONS.filter((option) => filterInvalidOptions(option.value as BoostType, type)),
+    [type]
+  );
+
+  return (
+    <EuiPageContent className="boosts">
+      <EuiPageContentHeader className="boosts__header" responsive={false}>
+        <EuiPageContentHeaderSection>
+          <EuiTitle size="xxs">
+            <h4>
+              {i18n.translate(
+                'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.boosts.title',
+                {
+                  defaultMessage: 'Boosts',
+                }
+              )}
+            </h4>
+          </EuiTitle>
+        </EuiPageContentHeaderSection>
+        <EuiPageContentHeaderSection>
+          <EuiSuperSelect
+            className="boosts__select"
+            itemClassName="boostSelectOption"
+            valueOfSelected={'add-boost'}
+            options={selectOptions}
+            onChange={(value) => addBoostClick(value as BoostType)}
```

Comment:
I'm confused as to why we need to declare a separate `addBoostClick` here. Either we should do this

```suggestion
            onChange={(value: BoostType) => addBoost(name, value)}
```

or we should do this

```suggestion
            onChange={addBoostClick}
```

slight preference to first option as you seem to be more consistently inlining fns in this PR

Reply:
a29ae47